runtime/server/batch_query.go

```diff
@@ -0,0 +1,177 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime/server/auth"
+)
+
+func (s *Server) QueryBatch(req *runtimev1.QueryBatchRequest, srv runtimev1.QueryService_QueryBatchServer) error {
+	ctx := srv.Context()
+
+	if !auth.GetClaims(ctx).CanInstance(req.InstanceId, auth.ReadOLAP) {
+		return ErrForbidden
+	}
+	var wg sync.WaitGroup
+
+	for _, query := range req.Queries {
+		wg.Add(1)
+		go func(query *runtimev1.QueryBatchSingleRequest) {
+			defer wg.Done()
+
+			resp := s.forwardQuery(ctx, query)
+
+			if err := srv.Send(resp); err != nil {
+				s.logger.Debug(fmt.Sprintf("Profiling Query Error: %v", err))
+			}
+		}(query)
+	}
+
+	wg.Wait()
+	return nil
+}
+
+func (s *Server) forwardQuery(ctx context.Context, query *runtimev1.QueryBatchSingleRequest) *runtimev1.QueryBatchResponse {
+	resp := &runtimev1.QueryBatchResponse{
+		Id:   query.Id,
+		Type: query.Type,
+	}
+
+	var err error
+	switch query.Type {
+	case runtimev1.QueryBatchType_QUERY_BATCH_TYPE_METRICS_VIEW_TOPLIST:
```

Comment:
Consider a type switch instead, so the enum isn't needed

Reply:
In all the cases, override the individual query's `instanceId` and `priority` to the value from the batched query. It paves the way for removing those fields from the individual queries. The repetition is okay for now

Reply:
Have updated to override `instanceId`. For `priority` does it make sense to only override if the priority is not passed?